hera_qm/omnical_metrics.py

```diff
@@ -12,3 +28,796 @@ def get_omnical_metrics_dict():
+                filename += '.json'
+
+            # change ndarrays to lists
+            metrics_out = copy.deepcopy(self.metrics)
+            for i, k in enumerate(metrics_out.keys()):
+                if type(metrics_out[k]) is np.ndarray:
+                    metrics_out[k] = metrics_out[k].tolist()
+                elif type(metrics_out[k]) is OrderedDict:
+                    for j in metrics_out[k].keys():
+                        if type(metrics_out[k][j]) is np.ndarray:
+                            metrics_out[k][j] = metrics_out[k][j].tolist()
+                elif type(metrics_out[k]) is np.bool_:
+                    metrics_out[k] = bool(metrics_out[k])
+
+            with open(filename, 'w') as f:
+                json.dump(metrics_out, f, indent=4)
+
+        elif filetype == 'pkl':
+            if filename.split('.')[-1] != 'pkl':
+                filename += '.pkl'
+            with open(filename, 'wb') as f:
+                outp = pkl.Pickler(f)
+                outp.dump(self.metrics)
+
+    def load_metrics(self, filename):
+        """
+        load omnical metrics
+
+        Input:
+        ------
+        filename : str
+            path to omnical metrics file
+
+        Result:
+        --------
+        self.metrics dictionary
+        """
+        self.metrics = load_omnical_metrics(filename)
+
+    def phs_FT_metric(self, phs_diff, phs_noise_cut=1.0, return_dict=True):
+        """
+        Takes the square of the real-valued FT of the phase difference between
+        omnical and fistcal solutions and uses it to assess noise level across freq
+
+        Input:
+        ------
+        phs_diff : ndarray, dtype=float, shape=(Nants, Ntimes, Nfreqs, Npols)
+            real ndarray containing difference between omnical gain phases
+            and firstcal gain phases
+
+        phs_noise_cut : float, default=1.0
+            phase noise level cut.
+            the noise level is estimated by taking the absolute value of the 
+            real-valued fourier transform of the omnical gain phase - firstcal
+            gain phase, taking a median filter across the modes, and then taking 
+            the median of the last 100 modes. 
+
+        return_dict : bool
+            return per-antenna output as dictionary with antenna name as key
+
+        Output:
+        -------
+        ant_phs_noise : ndarray, shape=(Nants,)
+            phs noise for each antenna
+
+        tot_phs_noise : float
+            average of ant_phs_noise
+
+        phs_noise_bad_ants : ndarray
+            array of antennas whose phs_noise is above the cut
+
+        phs_noise_good_sol : bool
+            boolean with the metric's good / bad determination for
+            the entire solution.
+        """
+        # take rfft
+        rfft = medfilt(np.abs(np.fft.rfft(phs_diff[:, :, self.band, :], axis=2)), kernel_size=(1, 1, 15, 1))
+
+        # Get phase noise
+        phs_noise = np.median(rfft[:, :, -100:, :], axis=2)
```

Comment:
I'm not a big fan of hardcoded numbers, here and the `15` above. Could these be programmatically determined based on the number of channels and/or bandwidth? Or at least optional input?